frontend/app/(modals)/comments.tsx

```diff
@@ -1,82 +1,376 @@
 import CommentComponent from '@/components/Comment';
-import { Container, Main } from '@/tamagui.config';
-import { Feather } from '@expo/vector-icons';
+import { UserContext } from '@/contexts/UserContext';
+import { CommentSchema, CommentSchemaInputs } from '@/types/commentInput';
+import {
+  AntDesign,
+  Feather,
+  MaterialCommunityIcons,
+  Octicons,
+} from '@expo/vector-icons';
+import { zodResolver } from '@hookform/resolvers/zod';
+import axios from 'axios';
 import { useLocalSearchParams, useRouter } from 'expo-router';
-import { Dimensions } from 'react-native';
-import { Button, ScrollView, Text, TextArea, XStack } from 'tamagui';
+import { useContext, useEffect, useState } from 'react';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import {
+  Dimensions,
+  KeyboardAvoidingView,
+  Modal,
+  useColorScheme,
+} from 'react-native';
+import GestureRecognizer from 'react-native-swipe-gestures';
+import { Button, Form, Input, ScrollView, Text, View } from 'tamagui';
 
 export default function CommentPage() {
-  const { name } = useLocalSearchParams();
+  //get user information
+  const { token, user_id } = useContext(UserContext);
+
+  //get parameters from the post
+  const item = useLocalSearchParams();
```

Comment:
You can add types to this so you can get autocomplete and Intellisense. If comments will always have the same parameters you can do something like `const {comments, post_id} = useLocalSearchParams<{comments : string, post_id: string}>();`
With this you can then do `JSON.parse(comments)` to get it back into the object form since it was passed in with `stringify`. 
I think that post_id is already a string so there is no need to `JSON.parse` that.